### PR TITLE
Refs #34900 -- Removed usage of deprecated glob.glob1().

### PR DESCRIPTION
**Pull request by earthyoung**
I removed usage of deprecated glob.glob1().

### CODE REVIEW

File: django/core/cache/backends/filebased.py [hunk 1]
```diff
@@ -164,7 +164,5 @@ def _list_cache_files(self):
         Get a list of paths to all the cache files. These are all the files
         in the root cache dir that end on the cache_suffix.
         """
-        return [
-            os.path.join(self._dir, fname)
-            for fname in glob.glob1(self._dir, "*%s" % self.cache_suffix)
-        ]
+        cache_suffix_paths = os.path.join(self._dir, f"*{self.cache_suffix}")
+        return glob.glob(cache_suffix_paths)
```
Comment (felixxm) [inline django/core/cache/backends/filebased.py:168]: Should not we use `root_dir` instead? :thinking: 
```suggestion
        return glob.glob(f"*{self.cache_suffix}", root_dir=self._dir)
```

Comment (earthyoung) [reply django/core/cache/backends/filebased.py:168]: That sounds good. I hadn't thought of that!

**Comment by felixxm** [comment]
`glob.glob1()` was deprecated in Python 3.13, https://docs.python.org/3/whatsnew/3.13.html#new-deprecations. We missed it somehow.

File: django/core/cache/backends/filebased.py [hunk 2]
```diff
@@ -164,7 +164,4 @@ def _list_cache_files(self):
         Get a list of paths to all the cache files. These are all the files
         in the root cache dir that end on the cache_suffix.
         """
-        return [
-            os.path.join(self._dir, fname)
-            for fname in glob.glob1(self._dir, "*%s" % self.cache_suffix)
-        ]
+        return glob.glob(f"*{self.cache_suffix}", root_dir=self._dir)
```
[the remark this answers is not in the document]
Comment (felixxm) [reply django/core/cache/backends/filebased.py:167]: Sorry for the wrong suggestion.

**Comment by felixxm** [comment]
> `glob.glob1()` was deprecated in Python 3.13, https://docs.python.org/3/whatsnew/3.13.html#new-deprecations. We missed it somehow.

Personally, I'd backport this to 5.1. We don't need a separate release note as `5.1.3.txt` already mentions Python 3.13: _"... and adds compatibility with Python 3.13."_.

**Comment by earthyoung** [comment]
That sounds good!

**Review by felixxm — APPROVED** [review]
@earthyoung Thanks for this patch :+1: Welcome aboard :boat:

**Review by sarahboyce — APPROVED** [review]
Thank you!

**Pull request merged**